build.psm1

```diff
@@ -393,15 +393,21 @@ cmd.exe /C cd /d "$location" "&" "$($vcVarsPath)\vcvarsall.bat" "$NativeHostArch
     if($PSModuleRestore)
     {
         # downloading the PackageManagement module
-        # $Options.Output is pointing to something like "...\src\powershell-win-core\bin\Debug\netcoreapp1.0\win10-x64\powershell.exe", 
+        # $Options.Output is pointing to something like "...\src\powershell-win-core\bin\Debug\netcoreapp1.0\win10-x64\publish\powershell.exe", 
         # so we need to get its parent directory
-        $PowerShellPath = Split-Path $Options.Output -Parent
-        if($Publish -and (-not $IsWindows))
+        $publishPath = Split-Path $Options.Output -Parent
+        log "Restore PowerShell modules to $publishPath"
+        $destPublishPath = Join-Path -Path $publishPath -ChildPath "Modules"
+        Restore-PSModule -Name PackageManagement -Destination $destPublishPath
+
+        # Strip publish because TravisCI needs it to run
+        $powerShellPath = Split-Path $publishPath -Parent
+        $destination = Join-Path -Path $powerShellPath -ChildPath "Modules"
+
+        if(Test-Path -path $destPublishPath)
```

Comment:
This logic is not correct

1. On incremental builds, it would Silently not change the folder
1. When user provide a custom `binDir`, it will pollute parent directory.

Please, consider fix `Start-PSPester` to understand both `-Publish` and no-publish options.
It could be as simple as add `-Publish` switch.